src/Microsoft.AspNetCore.Http.Abstractions/HostString.cs

```diff
@@ -38,6 +64,92 @@ public bool HasValue
         }
 
         /// <summary>
+        /// Returns the value of the domain part of the host. The port is removed if it was present.
+        /// IPv6 addresses will have brackets added if they are missing.
+        /// </summary>
+        /// <returns></returns>
+        public string Domain
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_value))
+                {
+                    return string.Empty;
+                }
+
+                // normalize the value
+                var uriComponent = ToUriComponent();
+
+                int index;
+                if ((index = uriComponent.IndexOf(']')) >= 0)
+                {
+                    // optimization if uriComponent is exactly the domain
+                    if(uriComponent.Length == index + 1)
+                    {
+                        return uriComponent;
+                    }
+
+                    // contains a port
+                    return uriComponent.Substring(0, index + 1);
+                }
+                else 
+                {
+                    // not IPV6
+                    if ((index = uriComponent.IndexOf(':')) >= 0)
+                    {
+                        // contains a port
+                        return uriComponent.Substring(0, index);
+                    }
+                    else
+                    {
+                        // no port
+                        return uriComponent;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Returns the value of the port part of the host, or <value>0<code> if none is found.
```

Comment:
Why doesn't this return `int?` and `null` in the no port case? That would align better with the new constructor.